api/src/repositories/telemetry-repositories/telemetry-vendor-repository.ts

```diff
@@ -315,4 +506,62 @@ export class TelemetryVendorRepository extends BaseRepository {
 
     return response.rows[0].count;
   }
+
+  /**
+   * Get telemetry record by telemetry ID.
+   *
+   * @param {number} surveyId
+   * @param {string} telemetryId
+   * @return {*}  {Promise<Telemetry>}
+   * @memberof TelemetryVendorRepository
+   */
+  async getTelemetryRecordById(surveyId: number, telemetryId: string): Promise<Telemetry> {
+    const knex = getKnex();
+
+    const queryBuilder = knex
+      .queryBuilder()
+      .with('telemetry', (withQueryBuilder) => {
+        withQueryBuilder.unionAll([
+          /**
+           * LOTEK Telemetry
+           */
+          knex
+            .queryBuilder()
+            .modify(this.getLotekTelemetryBaseQuery)
+            .modify(this.getLotekTelemetryBySurveyIdClause, surveyId)
+            .modify(this.getLotekTelemetryByTelemetryIdClause, telemetryId),
+          /**
+           * VECTRONIC Telemetry
+           */
+          knex
+            .queryBuilder()
+            .modify(this.getVectronicTelemetryBaseQuery)
+            .modify(this.getVectronicTelemetryBySurveyIdClause, surveyId)
+            .modify(this.getVectronicTelemetryByTelemetryIdClause, telemetryId),
+          /**
+           * ATS Telemetry
+           */
+          knex
+            .queryBuilder()
+            .modify(this.getATSTelemetryBaseQuery)
+            .modify(this.getATSTelemetryBySurveyIdClause, surveyId)
+            .modify(this.getATSTelemetryByTelemetryIdClause, telemetryId),
+          /**
+           * MANUAL Telemetry
+           */
+          knex
+            .queryBuilder()
+            .modify(this.getManualTelemetryBaseQuery)
+            .modify(this.getManualTelemetryBySurveyIdClause, surveyId)
+            .modify(this.getManualTelemetryByTelemetryIdClause, telemetryId)
+        ]);
+      })
+      .select('*')
+      .from('telemetry')
+      .where('telemetry.telemetry_id', telemetryId);
```

Comment:
question: Does this clause have any affect on this query?

Reply:
Hmm, possibly not

Reply:
Ya it could be removed